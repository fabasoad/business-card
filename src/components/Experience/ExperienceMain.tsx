import * as React from 'react'
import { useTranslation } from 'react-i18next'

import JobAchievements from './JobAchievements'
import JobDescription from './JobDescription'
import JobPeriod from './JobPeriod'
import JobTechnologies from './JobTechnologies'
import JobTitle from './JobTitle'

export default function ExperienceMain() {
  const { t } = useTranslation()
  return (
    <div id="experience" className="light-component">
      <div className="container">
        <div className="section-title text-center">
          <h2>{t('business-card-experience-title')}</h2>
          <hr />
        </div>
        <ul className="timeline">
          <li className="timeline-inverted">
            <JobPeriod
              fromMonth={9}
              fromYear={2020}
            />
            <div className="timeline-panel">
              <JobTitle
                company="TRI-AD"
                title={t('business-card-experience-job-tri-ad-2-title')}
              />
              <div className="timeline-body">
                <JobDescription
                  count={1}
                  i18nKeyPrefix="business-card-experience-job-tri-ad-2-description"
                />
                <JobTechnologies
                  keys={['concourseCI', 'kubernetes', 'prometheus', 'grafana']}
                />
              </div>
            </div>
          </li>
          <li>
            <JobPeriod
              fromMonth={5}
              fromYear={2020}
              toMonth={8}
              toYear={2020}
            />
            <div className="timeline-panel">
              <JobTitle
                company="TRI-AD"
                title={t('business-card-experience-job-tri-ad-1-title')}
              />
              <div className="timeline-body">
                <JobDescription
                  count={1}
                  i18nKeyPrefix="business-card-experience-job-tri-ad-1-description"
                />
                <JobTechnologies
                  keys={['docker', 'gitlab', 'gitlabCi', 'typeScript', 'react', 'maven', 'java', 'quarkus', 'vsCode', 'intellijIdea']}
                />
              </div>
            </div>
          </li>
          <li className="timeline-inverted">
            <JobPeriod
              fromMonth={7}
              fromYear={2019}
              toMonth={4}
              toYear={2020}
            />
            <div className="timeline-panel">
              <JobTitle
                company="Bitcoin.com"
                title={t('business-card-experience-job-bitcoin-com-2-title')}
              />
              <div className="timeline-body">
                <JobDescription
                  count={1}
                  i18nKeyPrefix="business-card-experience-job-bitcoin-com-2-description"
                />
                <JobAchievements
                  count={2}
                  i18nKeyPrefix="business-card-experience-job-bitcoin-com-2-achievements"
                />
                <JobTechnologies
                  keys={['docker', 'travis', 'gitHubActions', 'bitrise', 'java', 'kafka', 'spring', 'intellijIdea']}
                />
              </div>
            </div>
          </li>
          <li>
            <JobPeriod
              fromMonth={9}
              fromYear={2018}
              toMonth={6}
              toYear={2019}
            />
            <div className="timeline-panel">
              <JobTitle
                company="Bitcoin.com"
                title={t('business-card-experience-job-bitcoin-com-1-title')}
              />
              <div className="timeline-body">
                <JobDescription
                  count={1}
                  i18nKeyPrefix="business-card-experience-job-bitcoin-com-1-description"
                />
                <JobAchievements
                  count={1}
                  i18nKeyPrefix="business-card-experience-job-bitcoin-com-1-achievements"
                />
                <JobTechnologies
                  keys={[
                    'maven', 'java', 'tomcat', 'lombok', 'junit', 'python',
                    'javaScript', 'jQuery', 'bootstrap', 'mariaDb', 'redis',
                    'git', 'intellijIdea', 'vsCode', 'jira'
                  ]}
                />
              </div>
            </div>
          </li>
          <li className="timeline-inverted">
            <JobPeriod
              fromMonth={2}
              fromYear={2016}
              toMonth={9}
              toYear={2018}
            />
            <div className="timeline-panel">
              <JobTitle
                company="Lohika"
                title={t('business-card-experience-job-lohika-3-title')}
              />
              <div className="timeline-body">
                <JobDescription
                  count={2}
                  i18nKeyPrefix="business-card-experience-job-lohika-3-prysm-description"
                />
                <JobAchievements
                  count={2}
                  i18nKeyPrefix="business-card-experience-job-lohika-3-prysm-achievements"
                />
                <JobTechnologies
                  keys={['csharp', 'xamarin', 'prism', 'git', 'jira', 'teamCity', 'visualStudio']}
                />
                <p>&nbsp;</p>
                <JobDescription
                  count={2}
                  i18nKeyPrefix="business-card-experience-job-lohika-3-earnin-description"
                  withTitle={false}
                />
                <JobAchievements
                  count={6}
                  i18nKeyPrefix="business-card-experience-job-lohika-3-earnin-achievements"
                />
                <JobTechnologies
                  keys={[
                    'csharp', 'dotNetCore', 'kubernetes', 'helm', 'docker',
                    'elasticsearch', 'jenkins', 'kibana', 'fluentBit', 'fluentD',
                    'git', 'gitHub', 'prometheus', 'visualStudio', 'grafana'
                  ]}
                />
                <p>&nbsp;</p>
                <JobDescription
                  count={1}
                  i18nKeyPrefix="business-card-experience-job-lohika-3-general-description"
                  withTitle={false}
                />
                <JobAchievements
                  count={3}
                  i18nKeyPrefix="business-card-experience-job-lohika-3-general-achievements"
                />
              </div>
            </div>
          </li>
          <li>
            <JobPeriod
              fromMonth={5}
              fromYear={2013}
              toMonth={1}
              toYear={2016}
            />
            <div className="timeline-panel">
              <JobTitle
                company="Lohika"
                title={t('business-card-experience-job-lohika-2-title')}
              />
              <div className="timeline-body">
                <JobDescription
                  count={1}
                  i18nKeyPrefix="business-card-experience-job-lohika-2-description"
                />
                <JobAchievements
                  count={4}
                  i18nKeyPrefix="business-card-experience-job-lohika-2-achievements"
                />
                <JobTechnologies
                  keys={[
                    'java', 'spring', 'maven', 'javaScript', 'angular', 'mssql',
                    'oracle', 'svn', 'git', 'intellijIdea', 'hpeAlm', 'jenkins'
                  ]}
                />
              </div>
            </div>
          </li>
          <li className="timeline-inverted">
            <JobPeriod
              fromMonth={10}
              fromYear={2011}
              toMonth={4}
              toYear={2013}
            />
            <div className="timeline-panel">
              <JobTitle
                company="Lohika"
                title={t('business-card-experience-job-lohika-1-title')}
              />
              <div className="timeline-body">
                <JobDescription
                  count={1}
                  i18nKeyPrefix="business-card-experience-job-lohika-1-description"
                />
                <JobAchievements
                  count={3}
                  i18nKeyPrefix="business-card-experience-job-lohika-1-achievements"
                />
                <JobTechnologies
                  keys={['csharp', 'windowsForms', 'svn', 'visualStudio', 'hpeAlm', 'jenkins']}
                />
              </div>
            </div>
          </li>
          <li>
            <JobPeriod
              fromMonth={3}
              fromYear={2010}
              toMonth={9}
              toYear={2011}
            />
            <div className="timeline-panel">
              <JobTitle
                company="ARKA - Finance, Software, Services"
                title={t('business-card-experience-job-arka-title')}
              />
              <div className="timeline-body">
                <JobDescription
                  count={1}
                  i18nKeyPrefix="business-card-experience-job-arka-description"
                />
                <JobAchievements
                  count={3}
                  i18nKeyPrefix="business-card-experience-job-arka-achievements"
                />
                <JobTechnologies
                  keys={[
                    'php', 'codeigniter', 'vBulletin', 'oxidEShop', 'smarty', 'javaScript', 'jQuery',
                    'csharp', 'silverlight', 'mysql', 'svn', 'visualStudio', 'netBeans', 'redmine'
                  ]}
                />
              </div>
            </div>
          </li>
        </ul>
      </div>
    </div>
  )
}
