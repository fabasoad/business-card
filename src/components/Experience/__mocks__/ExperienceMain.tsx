import * as React from 'react'

export default function ExperienceMain() {
  return (
    <div data-testid="ExperienceMain"></div>
  )
}
