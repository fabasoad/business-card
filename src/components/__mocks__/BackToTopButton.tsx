import * as React from 'react'

export default function BackToTopButton() {
  return (
    <div data-testid="BackToTopButton"></div>
  )
}
