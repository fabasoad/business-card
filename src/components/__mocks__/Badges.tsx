import * as React from 'react'

export default function Badges() {
  return (
    <div data-testid="Badges"></div>
  )
}
