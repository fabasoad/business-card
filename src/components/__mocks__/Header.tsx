import * as React from 'react'

export default function Header() {
  return (
    <div data-testid="Header"></div>
  )
}
