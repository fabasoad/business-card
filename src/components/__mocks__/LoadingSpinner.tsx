import * as React from 'react'
import { LoadingSpinnerProps } from '../LoadingSpinner'

export default function LoadingSpinner({ background, color }: LoadingSpinnerProps) {
  return (
    <div data-testid={`LoadingSpinner-${background}-${color}`}></div>
  )
}
