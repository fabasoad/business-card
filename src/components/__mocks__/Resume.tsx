import * as React from 'react'

export default function Resume() {
  return (
    <div data-testid="Resume"></div>
  )
}
