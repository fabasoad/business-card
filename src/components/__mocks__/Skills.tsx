import * as React from 'react'

export default function Skills() {
  return (
    <div data-testid="Skills"></div>
  )
}
