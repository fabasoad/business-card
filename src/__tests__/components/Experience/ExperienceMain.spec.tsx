/// <reference types="jest" />
import * as React from 'react'
import { shallow } from 'enzyme'
import ExperienceMain from '../../../components/Experience/ExperienceMain'
import { useTranslation } from '../../__mocks__/react-i18next'

let tMock

beforeEach(() => {
  tMock = useTranslation().t
})

test('should render ExperienceMain correctly', () => {
  const wrapper = shallow(<ExperienceMain />)
  expect(wrapper).toMatchSnapshot()
  new Array<string>(
    'business-card-experience-title',
    'business-card-experience-job-tri-ad-2-title',
    'business-card-experience-job-tri-ad-1-title',
    'business-card-experience-job-bitcoin-com-2-title',
    'business-card-experience-job-bitcoin-com-1-title',
    'business-card-experience-job-lohika-3-title',
    'business-card-experience-job-lohika-2-title',
    'business-card-experience-job-lohika-1-title',
    'business-card-experience-job-arka-title'
  ).forEach((t, i) => expect(tMock).toHaveBeenNthCalledWith(i + 1, t))
})
