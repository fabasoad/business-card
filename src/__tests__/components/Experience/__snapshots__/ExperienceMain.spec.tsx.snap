// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExperienceMain correctly 1`] = `
<div
  className="light-component"
  id="experience"
>
  <div
    className="container"
  >
    <div
      className="section-title text-center"
    >
      <h2>
        business-card-experience-title
      </h2>
      <hr />
    </div>
    <ul
      className="timeline"
    >
      <li
        className="timeline-inverted"
      >
        <JobPeriod
          fromMonth={9}
          fromYear={2020}
        />
        <div
          className="timeline-panel"
        >
          <JobTitle
            company="TRI-AD"
            title="business-card-experience-job-tri-ad-2-title"
          />
          <div
            className="timeline-body"
          >
            <JobDescription
              count={1}
              i18nKeyPrefix="business-card-experience-job-tri-ad-2-description"
              withTitle={true}
            />
            <JobTechnologies
              keys={
                Array [
                  "concourseCI",
                  "kubernetes",
                  "prometheus",
                  "grafana",
                ]
              }
            />
          </div>
        </div>
      </li>
      <li>
        <JobPeriod
          fromMonth={5}
          fromYear={2020}
          toMonth={8}
          toYear={2020}
        />
        <div
          className="timeline-panel"
        >
          <JobTitle
            company="TRI-AD"
            title="business-card-experience-job-tri-ad-1-title"
          />
          <div
            className="timeline-body"
          >
            <JobDescription
              count={1}
              i18nKeyPrefix="business-card-experience-job-tri-ad-1-description"
              withTitle={true}
            />
            <JobTechnologies
              keys={
                Array [
                  "docker",
                  "gitlab",
                  "gitlabCi",
                  "typeScript",
                  "react",
                  "maven",
                  "java",
                  "quarkus",
                  "vsCode",
                  "intellijIdea",
                ]
              }
            />
          </div>
        </div>
      </li>
      <li
        className="timeline-inverted"
      >
        <JobPeriod
          fromMonth={7}
          fromYear={2019}
          toMonth={4}
          toYear={2020}
        />
        <div
          className="timeline-panel"
        >
          <JobTitle
            company="Bitcoin.com"
            title="business-card-experience-job-bitcoin-com-2-title"
          />
          <div
            className="timeline-body"
          >
            <JobDescription
              count={1}
              i18nKeyPrefix="business-card-experience-job-bitcoin-com-2-description"
              withTitle={true}
            />
            <JobAchievements
              count={2}
              i18nKeyPrefix="business-card-experience-job-bitcoin-com-2-achievements"
            />
            <JobTechnologies
              keys={
                Array [
                  "docker",
                  "travis",
                  "gitHubActions",
                  "bitrise",
                  "java",
                  "kafka",
                  "spring",
                  "intellijIdea",
                ]
              }
            />
          </div>
        </div>
      </li>
      <li>
        <JobPeriod
          fromMonth={9}
          fromYear={2018}
          toMonth={6}
          toYear={2019}
        />
        <div
          className="timeline-panel"
        >
          <JobTitle
            company="Bitcoin.com"
            title="business-card-experience-job-bitcoin-com-1-title"
          />
          <div
            className="timeline-body"
          >
            <JobDescription
              count={1}
              i18nKeyPrefix="business-card-experience-job-bitcoin-com-1-description"
              withTitle={true}
            />
            <JobAchievements
              count={1}
              i18nKeyPrefix="business-card-experience-job-bitcoin-com-1-achievements"
            />
            <JobTechnologies
              keys={
                Array [
                  "maven",
                  "java",
                  "tomcat",
                  "lombok",
                  "junit",
                  "python",
                  "javaScript",
                  "jQuery",
                  "bootstrap",
                  "mariaDb",
                  "redis",
                  "git",
                  "intellijIdea",
                  "vsCode",
                  "jira",
                ]
              }
            />
          </div>
        </div>
      </li>
      <li
        className="timeline-inverted"
      >
        <JobPeriod
          fromMonth={2}
          fromYear={2016}
          toMonth={9}
          toYear={2018}
        />
        <div
          className="timeline-panel"
        >
          <JobTitle
            company="Lohika"
            title="business-card-experience-job-lohika-3-title"
          />
          <div
            className="timeline-body"
          >
            <JobDescription
              count={2}
              i18nKeyPrefix="business-card-experience-job-lohika-3-prysm-description"
              withTitle={true}
            />
            <JobAchievements
              count={2}
              i18nKeyPrefix="business-card-experience-job-lohika-3-prysm-achievements"
            />
            <JobTechnologies
              keys={
                Array [
                  "csharp",
                  "xamarin",
                  "prism",
                  "git",
                  "jira",
                  "teamCity",
                  "visualStudio",
                ]
              }
            />
            <p>
               
            </p>
            <JobDescription
              count={2}
              i18nKeyPrefix="business-card-experience-job-lohika-3-earnin-description"
              withTitle={false}
            />
            <JobAchievements
              count={6}
              i18nKeyPrefix="business-card-experience-job-lohika-3-earnin-achievements"
            />
            <JobTechnologies
              keys={
                Array [
                  "csharp",
                  "dotNetCore",
                  "kubernetes",
                  "helm",
                  "docker",
                  "elasticsearch",
                  "jenkins",
                  "kibana",
                  "fluentBit",
                  "fluentD",
                  "git",
                  "gitHub",
                  "prometheus",
                  "visualStudio",
                  "grafana",
                ]
              }
            />
            <p>
               
            </p>
            <JobDescription
              count={1}
              i18nKeyPrefix="business-card-experience-job-lohika-3-general-description"
              withTitle={false}
            />
            <JobAchievements
              count={3}
              i18nKeyPrefix="business-card-experience-job-lohika-3-general-achievements"
            />
          </div>
        </div>
      </li>
      <li>
        <JobPeriod
          fromMonth={5}
          fromYear={2013}
          toMonth={1}
          toYear={2016}
        />
        <div
          className="timeline-panel"
        >
          <JobTitle
            company="Lohika"
            title="business-card-experience-job-lohika-2-title"
          />
          <div
            className="timeline-body"
          >
            <JobDescription
              count={1}
              i18nKeyPrefix="business-card-experience-job-lohika-2-description"
              withTitle={true}
            />
            <JobAchievements
              count={4}
              i18nKeyPrefix="business-card-experience-job-lohika-2-achievements"
            />
            <JobTechnologies
              keys={
                Array [
                  "java",
                  "spring",
                  "maven",
                  "javaScript",
                  "angular",
                  "mssql",
                  "oracle",
                  "svn",
                  "git",
                  "intellijIdea",
                  "hpeAlm",
                  "jenkins",
                ]
              }
            />
          </div>
        </div>
      </li>
      <li
        className="timeline-inverted"
      >
        <JobPeriod
          fromMonth={10}
          fromYear={2011}
          toMonth={4}
          toYear={2013}
        />
        <div
          className="timeline-panel"
        >
          <JobTitle
            company="Lohika"
            title="business-card-experience-job-lohika-1-title"
          />
          <div
            className="timeline-body"
          >
            <JobDescription
              count={1}
              i18nKeyPrefix="business-card-experience-job-lohika-1-description"
              withTitle={true}
            />
            <JobAchievements
              count={3}
              i18nKeyPrefix="business-card-experience-job-lohika-1-achievements"
            />
            <JobTechnologies
              keys={
                Array [
                  "csharp",
                  "windowsForms",
                  "svn",
                  "visualStudio",
                  "hpeAlm",
                  "jenkins",
                ]
              }
            />
          </div>
        </div>
      </li>
      <li>
        <JobPeriod
          fromMonth={3}
          fromYear={2010}
          toMonth={9}
          toYear={2011}
        />
        <div
          className="timeline-panel"
        >
          <JobTitle
            company="ARKA - Finance, Software, Services"
            title="business-card-experience-job-arka-title"
          />
          <div
            className="timeline-body"
          >
            <JobDescription
              count={1}
              i18nKeyPrefix="business-card-experience-job-arka-description"
              withTitle={true}
            />
            <JobAchievements
              count={3}
              i18nKeyPrefix="business-card-experience-job-arka-achievements"
            />
            <JobTechnologies
              keys={
                Array [
                  "php",
                  "codeigniter",
                  "vBulletin",
                  "oxidEShop",
                  "smarty",
                  "javaScript",
                  "jQuery",
                  "csharp",
                  "silverlight",
                  "mysql",
                  "svn",
                  "visualStudio",
                  "netBeans",
                  "redmine",
                ]
              }
            />
          </div>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
